src/transaction/types/witness.rs

```diff
@@ -73,12 +76,20 @@ impl Distribution<Witness> for Standard {
     }
 }
 
+impl bytes::SizedBytes for Witness {
+    fn serialized_size(&self) -> usize {
+        WORD_SIZE + bytes::padded_len(self.data.as_slice())
+    }
+}
+
+#[cfg(feature = "std")]
 impl io::Read for Witness {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
         bytes::store_bytes(buf, self.data.as_slice()).map(|(n, _)| n)
```

Comment:
While our standard interfaces for serialization use std::io, none of this logic seems like it should require `std`. It's just converting a slice of bytes into an array.

Reply:
Not saying that should be done in this PR, but maybe a good task for people looking for open source contribution work.

Reply:
Blanket implementations would probably bring the need to copy on serialization. It will involve a complicated scheme to have a blanket zero-copy `io` implementation

Reply:
Looking at `fuel-types::bytes` it seems like they could be easily made no-std compatible if we just used a different error type. Resolving this for now since it seems like something we could fix in fuel-types.